extensions-core/kafka-extraction-namespace/src/main/java/org/apache/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -297,6 +297,18 @@ public LookupIntrospectHandler getIntrospectHandler()
     return new KafkaLookupExtractorIntrospectionHandler(this);
   }
 
+  @Override
+  public void awaitInitialization()
+  {
+    // Kafka lookup do not need await on initialization as it is realtime kafka lookups.
```

Comment:
For first time, KafkaLookupExtractorFactory has lifecycle and process will wait for started latch in start method. 
Today, On updating the config, it deletes the old lookup immediately and it kicks the start lifecycle again.  I just wanted to keep the same behavior as this change was mainly to cover jdbc use case, thus overridden await methods. 

But you have great point, I think we should await for kafka, reading from kafka and populating is asynchronus and ideally it should await on initialization as well to avoid the disruption in the looking serving requests. 
Let me remove the await methods so that it will wait on new lookup to get populated from new/old topic and await on init. 
**New behavior will be:** 
kafka lookup will wait until loadTimeout to drain all messages from kafka topic and it will be initialized fully and then it will drop the old container and starting serving from new one, else it will continue serving from old lookup. 

Thank you for comment.

Reply:
i think its fine to do as a follow-up, thanks for looking into it